### PR TITLE
factory_method_#1

### PR DESCRIPTION
**Pull request by MichaelAndrew22**
_Basic Implementation and demonstration of the factory method pattern._
Prototype #1 (can easily be improved)

 To design the led_strip classes we need to locate led_strip methods, (which I can't locate in diozero), eyes needed on locating led_strip methods in order to make these strip classes that the factory then instantiates based on string input.

Also, this is the new repo we are working from right? If so can we get the issues going on this bad boy?
Thanks for all your help :)

### CODE REVIEW

File: pidev/src/main/java/models/Device.java [hunk 1]
```diff
@@ -1,10 +1,10 @@
 package models;
 
 public class Device {
-    private String deviceName;
```
Comment (MichaelAndrew22) [inline pidev/src/main/java/models/Device.java:4]: Might want to keep these private? 
Is there a way to merge everything but this?

File: pidev/src/main/java/led/lightstrip/scrum5/pidev/PidevApplication.java [hunk 2]
```diff
@@ -1,13 +1,30 @@
...
 
+
+
+
 	public static void main(String[] args) {
 		SpringApplication.run(PidevApplication.class, args);
+
+
+		// testing factory
+		Device myDevice = new factory().createDevice("neopixel");
```
Comment (MichaelAndrew22) [inline pidev/src/main/java/led/lightstrip/scrum5/pidev/PidevApplication.java:18]: Can change this to use parameter passed into factory. In the constructor createDevice() can happen.

**Review by MichaelAndrew22 — COMMENTED** [review]
Some things can be polished but I think this is a pretty basic framework of the factory method in action.

**Review by Cxrdary — CHANGES_REQUESTED** [review]
looks good, merge into the dev branch please

**Comment by MichaelAndrew22** [comment]
Check this out, I think this is ultimately what he's going to teach us and then we are going to adopt it.

https://pi4j.com/getting-started/javafx-mvc-template/

**Pull request closed without merging**